scipy/linalg/flapack_pos_def_tri.pyf.src

```diff
@@ -102,15 +102,14 @@ subroutine <prefix>pteqr(compute_z, n, d, e, z, ldz, work, info)
     ! tridiagonal form, however, may preclude the possibility of obtaining
     ! high relative accuracy in the small eigenvalues of the original
     ! matrix, if these eigenvalues range over many orders of magnitude.)
-    callstatement (*f2py_func)(compute_z, &n, d, e, z, &ldz, work, &info)
+    callstatement (*f2py_func)((compute_z?(compute_z==2?"I":"V"):"N"), &n, d, e, z, &ldz, work, &info)
     callprotoargument char*, int*, <ctypereal>*, <ctypereal>*, <ctype>*, int*, <ctypereal>*, int*
-
-    character intent(in, optional), check(*compute_z=='N'||*compute_z=='V'||*compute_z=='I') :: compute_z='N'
+    character intent(in, optional), check((compute_z>=0) && (compute_z<=2)) :: compute_z = 0
```

Comment:
> Based on the error messages, I would guess that the wrapper thinks it's a string. - @mdhaber

It does. You need to tell it to expect an integer **not** a character.

```suggestion
    integer intent(in, optional), check((compute_z>=0) && (compute_z<=2)) :: compute_z = 0
```

Reply:
Thanks. As you know, I'm not as familiar with these; I only did one very simple one.

Now I'm getting:
```
____________________________________________ test_pteqr[0-float32-float32] ____________________________________________
scipy\linalg\tests\test_lapack.py:2133: in test_pteqr
    d_pteqr, e_pteqr, z_pteqr, info = pteqr(d=d, e=e, z=z, compute_z=compute_z)
```
because:
```
<ftype> intent(in,out,copy), depend(n), dimension((compute_z==0?1:max(1,n)),(compute_z==0?0:n)) :: z
```
If `compute_z`==0, I think we don't care what the dimensions of `z` are; the documentation states `If  COMPZ = 'N', then Z is not referenced.` So I modified that line:
```
<ftype> intent(in,out,copy), depend(n), dimension((compute_z==0?shape(z, 0):max(1,n)),(compute_z==0?shape(z, 1):n)) :: z
```

Reply:
I'm not sure where you're getting errors from. All tests on the CI pass (the failing test is from differential evolution) as they do on my machine.